src/main/java/org/elasticsearch/search/MultiValueMode.java

```diff
@@ -274,6 +315,34 @@ protected int pick(RandomAccessOrds values) {
                 return -1;
             }
         }
+    },
+
+    /**
+     * Count of values for the field per document.
+     *
+     * Note that apply/reduce do not work with COUNT since count cannot be derived from
+     * an accumulator algorithm without a consistency between which value is the accumulation.
+     */
+    COUNT {
```

Comment:
Maybe this one would be best outside of MultiValueMode since it is quite different from the other ones in that it always produces integers no matter what the values source is?